src/matlab2tikz.m

```diff
@@ -435,11 +435,11 @@ function matlab2tikz(varargin)
 
     % Alternative Positioning of axes.
     % Select relevant Axes and draw them.
-    [relevantAxesHandles, axesBoundingBox] = getRelevantAxes(m2t, axesHandles);
+    [m2t, axesBoundingBox] = getRelevantAxes(m2t, axesHandles);
```

Comment:
By looking at the code I don't see where you need the `relevantAxesHandles` in `drawBackgroundOfAxes()`.  Consequently it is not necessary to store the `relevantAxesHandles` in the `m2t` struct. You check the handle's visibility by directly calling `isVisible()` with the current `handle`.